mod_reforged/hooks/entity/tactical/enemies/schrat_small.nut

```diff
@@ -41,4 +41,23 @@
 		this.m.Skills.add(::new("scripts/skills/actives/rf_schrat_small_root_skill"));
 		this.m.IsActingImmediately = true;
 	}
+
+	// Return 0 during the call from onOtherActorDeath when using my own skill to kill myself so that my death
+	// in this case does not trigger morale checks because vanilla returns early in onOtherActorDeath if `getXPValue() <= 1`
+	q.getXPValue = @(__original) function()
+	{
+		local infos = ::getstackinfos(2);
+		if (split(infos.src, "/").top() == "actor.nut")
+		{
+			// Just checking for infos.func is not enough because hooked functions appear as "unknown"
```

Comment:
On top of that function being "unknown", it might also not even be "onOtherActorDeath".
Instead the stack, we are viewing, could be that of a hooked `getXPValue` function, if it was hooked twice by active mods.

Reply:
I see, yeah that's a concern. Although, in this particular case I think it's very unlikely that someone will hook `getXPValue()` of `schrat_small`. Maybe they will do a `hookTree` on `actor` for `getXPValue()` but that also seems low probability to me.

We can wait and try to think of more elegant solutions.